behave/contrib/csv_table_from_file.py

```diff
@@ -53,25 +93,43 @@ def read_examples_table_data_from_csv(example, file_path):
                 example.table.add_row(row)
 
 
-def preprocess_and_read_examples_table_data_from_csv(feature):
+def preprocess_and_read_examples_table_data_from_csv(feature: Feature):
     """
     Preprocesses and reads data from CSV files for scenario outlines in the given feature.
 
+    This function iterates over all scenario outlines in the provided feature,
+    looking for Examples tables tagged with @from_file={csv_filename}. For each
+    such tag, it reads data from the specified CSV file and populates the Examples
+    table with the data.
+
     Args:
-        feature (behave.model.Feature): The Feature object containing scenario outlines.
+        feature (Feature): The Feature object containing scenario outlines.
 
     Logs:
-        Logs warnings if CSV files for scenario outlines are not found.
+        Logs warnings if CSV files for scenario outlines are not found or if the files are not CSV files.
     """
-    logger.info(f"Feature filename: {feature.filename}")
+    logger.info(f"Processing feature file: {feature.filename}")
     for scenario_outline in feature.iter_scenario_outlines():
-        base_path = os.path.splitext(feature.filename)[0]
-        csv_file_path = f"{base_path}.csv"
-
-        if not os.path.exists(csv_file_path):
-            logger.warning(f"CSV file not found: {csv_file_path}")
-            continue
-
         for example in scenario_outline.examples:
-            read_examples_table_data_from_csv(example, csv_file_path)
-
+            csv_file_path = None
+
+            # Find the @from_file={csv_filename} tag
+            for tag in example.tags:
```

Comment:
CANDIDATE FOR HELPER-FUNCTION: `select_marker_tag_and_extract_filename(tags) -> Optional[str]`